website/statistics.py

```diff
@@ -121,6 +221,59 @@ def get_program_stats(self, user):
         }
         return jsonify(response)
 
+    def get_grid_info(self, user, class_id, level):
+        class_ = self.db.get_class(class_id)
+        if hedy_content.Adventures(g.lang).has_adventures():
+            adventures = hedy_content.Adventures(g.lang).get_adventure_keyname_name_levels()
+        else:
+            adventures = hedy_content.Adventures("en").get_adventure_keyname_name_levels()
+
+        students = sorted(class_.get("students", []))
+        teacher_adventures = self.db.get_teacher_adventures(user["username"])
+        class_info = self.db.get_class_customizations(class_id)
+        class_adventures = class_info.get('sorted_adventures')
+
+        adventure_names = {}
+        for adv_key, adv_dic in adventures.items():
+            for name, _ in adv_dic.items():
+                adventure_names[adv_key] = name
+
+        for adventure in teacher_adventures:
+            adventure_names[adventure['id']] = adventure['name']
+
+        class_adventures_formatted = {}
+        for key, value in class_adventures.items():
+            adventure_list = []
+            for adventure in value:
+                adventure_list.append(adventure_names[adventure['name']])
+            class_adventures_formatted[key] = adventure_list
+
+        ticked_adventures = {}
+        student_adventures = {}
+        for student in students:
```

Comment:
We can also try to reduce the nesting here a little bit as well?

Reply:
This does two things indeed and can be split. However, each split will be almost identical in terms of what is being iterated through but the returned variables will be the same. I think it could've been done in a less complex way but refactoring this means refactoring a lot of the code.